src/main/resources/api/service_openapi.yaml

```diff
@@ -62,6 +62,46 @@ paths:
         500:
           description: Get request error
           $ref: '#/components/responses/ErrorResponse'
+  '/api/v1/workspaces/{id}/references/data':
```

Comment:
these paths were how i interpreted the design doc. "uncontrolled resource" seemed to be more of an internal concept, and not something to expose at the API level. references and resources seemed like the two key things to expose at this level.

i'm imagining once controlled resources are implemented, the paths might look something like:

```
/api/v1/workspaces/{workspaceId}/references/data/{name}             //uncontrolled
/api/v1/workspaces/{workspaceId}/resources/data/{name}              //controlled
/api/v1/workspaces/{workspaceId}/resources/compute/{name}           //controlled
etc...
```

let me know if you disagree with the thinking.

Reply:
I think there's definitely some confusing parts from the design doc - we have one table for data references and one table for workspace resources, but those aren't orthogonal. E.g.:

uncontrolled data references - in data_references, not in workspace_resources
controlled data references - in both data_references and workspace_resources (linked via resource_id)
controlled compute resources - not in data_references, in workspace_resources

From a user perspective though, it seems like we'd want one endpoint to return data references and one endpoint for returning resources (though controlled data references could show up in both). E.g. if I want to get info about a data reference, it seems strange to call different endpoints depending on whether it's controlled or uncontrolled.

I think an alternative path structure could be something like
```
/api/v1/workspaces/{workspaceId}/datareferences   // data resources, both controlled + uncontrolled
/api/v1/workspaces/{workspaceId}/resources // controlled resources, including both data + compute
```
with param-based filtering for get/list requests.

Definitely not 100% on this idea though, I'd love to hear thoughts on this